test/uptech_growthbook_sdk_flutter_test.dart

```diff
@@ -102,26 +102,51 @@ void main() {
       });
 
       group('when the attribute meets the condition', () {
-        setUp(() {
-          const String greaterThan = '\$gt';
-          ToglTest.instance.initForTests(
-            seeds: {
-              'some-feature': false,
-            },
-            rules: [
-              {
-                'condition': {
-                  'version': {greaterThan: '1.0.0'}
-                },
-                'force': true
-              }
-            ],
-            attributes: {'version': '1.0.1'},
-          );
+        group('and the attribute is added on init', () {
+          setUp(() {
+            const String greaterThan = '\$gt';
+            ToglTest.instance.initForTests(
+              seeds: {
+                'some-feature': false,
+              },
+              rules: [
+                {
+                  'condition': {
+                    'version': {greaterThan: '1.0.0'}
+                  },
+                  'force': true
+                }
+              ],
+              attributes: {'version': '1.0.1'},
+            );
+          });
+
+          test('it returns true', () {
+            expect(ToglTest.instance.isOn('some-feature'), isTrue);
+          });
         });
+        group('and the attribute is added asynchronously', () {
+          setUp(() {
+            const String greaterThan = '\$gt';
+            ToglTest.instance.initForTests(
+              seeds: {
+                'some-feature': false,
+              },
+              rules: [
+                {
+                  'condition': {
+                    'version': {greaterThan: '1.0.0'}
+                  },
+                  'force': true
+                }
+              ],
+            );
+            ToglTest.instance.setAttributes({'version': '1.0.1'});
```

Comment:
You would need to add an `await` here as well to make it actually block until it is done setting it. Otherwise you would have a race condition.

The fact that the test is working as is makes me believe also that the `refresh()` above is not necessary.